timemachine/md/enhanced.py

```diff
@@ -553,3 +574,90 @@ def aligned_batch_propose(xvb, K, key, vacuum_samples, vacuum_log_weights):
 def jax_aligned_batch_propose_coords(x, K, key, vacuum_samples, vacuum_log_weights):
     vacuum_samples = jax_sample_from_log_weights(vacuum_samples, vacuum_log_weights, K, key)
     return batch_align_and_replace(vacuum_samples, x)
+
+
+def pregenerate_samples(mol, ff, seed, n_solvent_samples=1000, n_ligand_batches=30000, temperature=300.0, pressure=1.0):
+    potentials, params, masses, coords, box = get_solvent_phase_system(mol, ff)
+    print(f"Generating {n_solvent_samples} solvent samples")
+    solvent_xvbs = generate_solvent_samples(
+        coords, box, masses, potentials, params, temperature, pressure, seed, n_solvent_samples
+    )
+
+    print("Generating ligand samples")
+    ligand_samples, ligand_log_weights = generate_ligand_samples(n_ligand_batches, mol, ff, temperature, seed)
+
+    return solvent_xvbs, ligand_samples, ligand_log_weights
+
+
+def load_or_pregenerate_samples(
+    mol, ff, seed, n_solvent_samples=1000, n_ligand_batches=30000, temperature=300.0, pressure=1.0
+):
+
+    # hash canonical smiles
+    mol_hash = hash_mol(mol)
+    # TODO: do I want this to be sensitive to other mol properties? conformer?
+
+    # hash all of the other parameters
+    # ff_hash = hash(tuple(np.array(ff.get_ordered_params()).flatten()))
+    # arg_hash = hash((seed, n_solvent_samples, n_ligand_batches, temperature, pressure))
+    # TODO: do I want this to be sensitive to ff's smirks strings too?
+    # TODO: store and check ff_hash, arg_hash match the one stored...
+
+    cache_path = PATH_TO_SAMPLE_CACHES / f"mol_{mol_hash}_x_solvent_samples.pkl"
```

Comment:
Is it possible to generate a hash of the force field to include in the cache key?

Reply:
Removed for now in https://github.com/proteneer/timemachine/pull/718/commits/fea2af2b93d4b18cb6d53be11d7562dd7139d72e , but I think that would be a good idea